### DISCUSSION TITLE
Could really use some guidance upgrading Netbox from v2.4.4 to current

### DISCUSSION THREAD

**Discussion by quantumzedno** [netbox-community/netbox#8396] [Q&A] [answered]
Hi All,

I could really use some assistance, I started a new position and the company i'm working for now is already leveraging netbox for IPAM/DCIM, however their instance hasnt been updated in years... it's currently sitting on version 2.4.4 and i'm having a heck of a time trying to update it.

I cloned the VM so i can test trying to upgrade it without corrupting anything, but after installing updated versions of all the dependencies listed on the upgrade doc, i ran the upgrade script and got the following error dump. 

>  sudo PYTHON=/usr/bin/python3.8 ./upgrade.sh
Running NetBox upgrade as root, press any key to continue or ^C to cancel
Cleaning up stale Python bytecode (find . -name "*.pyc" -delete)...
Removing old Python packages (pip3 uninstall -r old_requirements.txt -y)...
WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
WARNING: Skipping django-rest-swagger as it is not installed.
WARNING: Skipping psycopg2 as it is not installed.
WARNING: Skipping pycrypto as it is not installed.
WARNING: Running pip as root will break packages and permissions. You should install packages reliably by using venv: https://pip.pypa.io/warnings/venv
Updating required Python packages (pip3 install -r requirements.txt --upgrade)...
WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
Collecting Django<2.1,>=1.11
  Downloading Django-2.0.13-py3-none-any.whl (7.1 MB)
     |████████████████████████████████| 7.1 MB 9.1 MB/s
Collecting django-cors-headers==2.4.0
  Using cached django_cors_headers-2.4.0-py2.py3-none-any.whl (12 kB)
Collecting django-debug-toolbar==1.9.1
  Using cached django_debug_toolbar-1.9.1-py2.py3-none-any.whl (206 kB)
Collecting django-filter==1.1.0
  Using cached django_filter-1.1.0-py2.py3-none-any.whl (45 kB)
Collecting django-mptt==0.9.1
  Using cached django_mptt-0.9.1-py2.py3-none-any.whl (105 kB)
Collecting django-tables2==1.21.2
  Using cached django-tables2-1.21.2.tar.gz (2.7 MB)
Collecting django-taggit==0.22.2
  Using cached django_taggit-0.22.2-py2.py3-none-any.whl (45 kB)
Collecting django-taggit-serializer==0.1.7
  Using cached django-taggit-serializer-0.1.7.tar.gz (5.6 kB)
Collecting django-timezone-field==2.1
  Using cached django_timezone_field-2.1-py2.py3-none-any.whl (8.4 kB)
Collecting djangorestframework==3.8.1
  Using cached djangorestframework-3.8.1-py2.py3-none-any.whl (923 kB)
Collecting drf-yasg[validation]==1.9.2
  Using cached drf_yasg-1.9.2-py2.py3-none-any.whl (841 kB)
Collecting graphviz==0.8.4
  Using cached graphviz-0.8.4-py2.py3-none-any.whl (16 kB)
Collecting Markdown==2.6.11
  Using cached Markdown-2.6.11-py2.py3-none-any.whl (78 kB)
Collecting natsort==5.3.3
  Using cached natsort-5.3.3-py2.py3-none-any.whl (34 kB)
Collecting ncclient==0.6.0
  Using cached ncclient-0.6.0.tar.gz (69 kB)
Collecting netaddr==0.7.19
  Using cached netaddr-0.7.19-py2.py3-none-any.whl (1.6 MB)
Collecting paramiko==2.4.1
  Using cached paramiko-2.4.1-py2.py3-none-any.whl (194 kB)
Collecting Pillow==5.2.0
  Downloading Pillow-5.2.0.tar.gz (14.5 MB)
     |████████████████████████████████| 14.5 MB 71.5 MB/s
Collecting psycopg2-binary==2.7.5
  Downloading psycopg2-binary-2.7.5.tar.gz (427 kB)
     |████████████████████████████████| 427 kB 41.2 MB/s
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-3guid84m/psycopg2-binary_e4d8eed0bc3b44449db4dcedea8fa08d/setup.py'"'"'; __file__='"'"'/tmp/pip-install-3guid84m/psycopg2-binary_e4d8eed0bc3b44449db4dcedea8fa08d/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-supy5x2o
         cwd: /tmp/pip-install-3guid84m/psycopg2-binary_e4d8eed0bc3b44449db4dcedea8fa08d/
    Complete output (23 lines):
    running egg_info
    creating /tmp/pip-pip-egg-info-supy5x2o/psycopg2_binary.egg-info
    writing /tmp/pip-pip-egg-info-supy5x2o/psycopg2_binary.egg-info/PKG-INFO
    writing dependency_links to /tmp/pip-pip-egg-info-supy5x2o/psycopg2_binary.egg-info/dependency_links.txt
    writing top-level names to /tmp/pip-pip-egg-info-supy5x2o/psycopg2_binary.egg-info/top_level.txt
    writing manifest file '/tmp/pip-pip-egg-info-supy5x2o/psycopg2_binary.egg-info/SOURCES.txt'

    Error: pg_config executable not found.

    pg_config is required to build psycopg2 from source.  Please add the directory
    containing pg_config to the $PATH or specify the full executable path with the
    option:

        python setup.py build_ext --pg-config /path/to/pg_config build ...

    or with the pg_config option in 'setup.cfg'.

    If you prefer to avoid building psycopg2 from source, please install the PyPI
    'psycopg2-binary' package instead.

    For further information please check the 'doc/src/install.rst' file (also at
    <http://initd.org/psycopg/docs/install.html>).

    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/17/dc/3f6bfae668594013cb9ef0cba4f4863451220944650868ad67332ca9069c/psycopg2-binary-2.7.5.tar.gz#sha256=c2ac7aa1a144d4e0e613ac7286dae85671e99fe7a1353954d4905629c36b811c (from https://pypi.org/simple/psycopg2-binary/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement psycopg2-binary==2.7.5 (from versions: 2.7.4, 2.7.5, 2.7.6, 2.7.6.1, 2.7.7, 2.8, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.8.5, 2.8.6, 2.9, 2.9.1, 2.9.2, 2.9.3)
ERROR: No matching distribution found for psycopg2-binary==2.7.5
WARNING: You are using pip version 21.1.1; however, version 21.3.1 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
Applying database migrations (python3 netbox/manage.py migrate)...
Traceback (most recent call last):
  File "netbox/manage.py", line 8, in <module>
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'
Collecting static files (python3 netbox/manage.py collectstatic --no-input)...
Traceback (most recent call last):
  File "netbox/manage.py", line 8, in <module>
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'

> 

i'm assuming i need to do some kind of interm upgrade before moving to the latest version but i don't know where to start.  

I have spun up an install of the latest release on a separate VM, so ideally id love to just do any Database upgrades and import the old database into the new server if possible, but again, i don't know where to start with something like that either.

Thank you so much in advance

**Comment by jeremystretch** [comment, marked as answer]
For something this far out of date, your best course of action is probably going to be:

1. [Make a copy of the database and media archive](https://netbox.readthedocs.io/en/stable/administration/replicating-netbox/) (you can skip the media archive if your installation doesn't have any image attachments)
2. Create an entirely _new_ installation of NetBox v2.11.12 and make sure that it works
3. Drop the database from the v2.11.12 install and load your copy from v2.4
4. Run `upgrade.sh` to apply all database migrations
5. Restart the NetBox services and verify that the data was loaded successfully
6. Upgrade to the latest v3.1 release

**Reply by quantumzedno** [reply]
Thank you so much Jeremy, that worked great! really appreciate the assistance and looking forward to trying out all the new features in netbox/the plugin system

**Reply by dayonoiki** [reply]
> For something this far out of date, your best course of action is probably going to be:
> 
> 1. [Make a copy of the database and media archive](https://netbox.readthedocs.io/en/stable/administration/replicating-netbox/) (you can skip the media archive if your installation doesn't have any image attachments)
> 2. Create an entirely _new_ installation of NetBox v2.11.12 and make sure that it works
> 3. Drop the database from the v2.11.12 install and load your copy from v2.4
> 4. Run `upgrade.sh` to apply all database migrations
> 5. Restart the NetBox services and verify that the data was loaded successfully
> 6. Upgrade to the latest v3.1 release

Hi, 
   I am in an exact same position with my new employer but in my instance, the current version is 1.9.2 and looking to upgrade to a later version (3.2.3). I do not even see 1.9.2 listed on the release versions on netbox documentation and I can only assume this will be a major release feature change. Any guidance on how to upgrade from 1.9.2 as have no idea where to start.
I

**Reply by candlerb** [reply]
It should be exactly the same process. Export the database; create a fresh v2.11.12 instance, import the database, upgrade; repeat for v3.2.5.

Note that various dependencies are now higher (especially Python 3.8+, Postgres 10+). Hence I strongly suggest building this from scratch on a completely new system or VM - don't attempt to upgrade in-place.

The other thing to beware is that there are significant API and data model changes between Netbox releases, so if you have any systems that integrate with Netbox's REST API then they need testing and are likely to require changes.

**Reply by dayonoiki** [reply]
Hi thanks for you response. 
Over the weekend, I created a new netbox v2.11.12 and netbox install was successful and working  (postgreSQL 12.11 , django 4.0.6 , Python 3.8.10). Then I dropped the database and loaded the netbox v1.9.2 database. 

All was loading well until we hit what looks like an SQL error while applying extras standardize models. Any assistance on what this may be ? and how to fix can this be fixed? 

In the  \ds and \dt there is a mix of owners for the different names. Unfortunately cannot see which of the names this failed field applies to from the list. see attachment

[dt_ds-log.txt](https://github.com/netbox-community/netbox/files/9084107/dt_ds-log.txt)
[log-output.txt](https://github.com/netbox-community/netbox/files/9084128/log-output.txt)

log outputs below : 
Applying dcim.0088_powerfeed_available_power... OK
  Applying dcim.0089_deterministic_ordering... OK
  Applying dcim.0090_cable_termination_models... OK
  Applying dcim.0091_interface_type_other... OK
  Applying dcim.0092_fix_rack_outer_unit... OK
  Applying dcim.0093_device_component_ordering... OK
  Applying dcim.0094_device_component_template_ordering... OK
  Applying dcim.0095_primary_model_ordering... OK
  Applying dcim.0096_interface_ordering... OK
  Applying dcim.0097_interfacetemplate_type_other... OK
  Applying dcim.0098_devicetype_images... OK
  Applying dcim.0099_powerfeed_negative_voltage... OK
  Applying dcim.0100_mptt_remove_indexes... OK
  Applying dcim.0101_nested_rackgroups... OK
  Applying dcim.0102_nested_rackgroups_rebuild... OK
  Applying dcim.0103_standardize_description... OK
  Applying dcim.0104_correct_infiniband_types... OK
  Applying dcim.0105_interface_name_collation... OK
  Applying dcim.0106_role_default_color... OK
  Applying dcim.0107_component_labels... OK
  Applying dcim.0108_add_tags... OK
  Applying dcim.0109_interface_remove_vm... OK
  Applying dcim.0110_virtualchassis_name... OK
  Applying dcim.0111_component_template_description... OK
  Applying dcim.0112_standardize_components... OK
  Applying dcim.0113_nullbooleanfield_to_booleanfield... OK
  Applying dcim.0114_update_jsonfield... OK
  Applying dcim.0115_rackreservation_order... OK
  Applying dcim.0116_rearport_max_positions... OK
  Applying dcim.0117_custom_field_data... OK
  Applying circuits.0016_3569_circuit_fields... OK
  Applying circuits.0017_circuittype_description... OK
  Applying circuits.0018_standardize_description... OK
  Applying circuits.0019_nullbooleanfield_to_booleanfield... OK
  Applying circuits.0020_custom_field_data... OK
  Applying extras.0051_migrate_customfields... OK
  Applying extras.0052_customfield_cleanup... OK
  Applying extras.0053_rename_webhook_obj_type... OK
  ****Applying extras.0054_standardize_models...Traceback (most recent call last):
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.ObjectNotInPrerequisiteState: sequence must have same owner as table it is linked to****


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "netbox/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/migrations/operations/fields.py", line 244, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 608, in alter_field
    self._alter_field(model, old_field, new_field, old_type, new_type,
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/postgresql/schema.py", line 196, in _alter_field
    super()._alter_field(
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 794, in _alter_field
    self.execute(sql, params)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 145, in execute
    cursor.execute(sql, params)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/cacheops/transaction.py", line 97, in execute
    result = self._no_monkey.execute(self, sql, params)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/opt/netbox-2.11.12/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.OperationalError: sequence must have same owner as table it is linked to  


nisadmin@netbox-test:/opt/netbox/netbox$ python3 manage.py sqlmigrate extras 0054
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/dist-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/dist-packages/django/core/management/__init__.py", line 420, in execute
    django.setup()
  File "/usr/local/lib/python3.8/dist-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.8/dist-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/lib/python3.8/dist-packages/django/apps/config.py", line 228, in create
    import_module(entry)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'cacheops'
nisadmin@netbox-test:/opt/netbox/netbox$

nisadmin@netbox-test:/opt/netbox/netbox$ python3 manage.py showmigrations
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/dist-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/dist-packages/django/core/management/__init__.py", line 420, in execute
    django.setup()
  File "/usr/local/lib/python3.8/dist-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.8/dist-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/lib/python3.8/dist-packages/django/apps/config.py", line 228, in create
    import_module(entry)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'cacheops'
nisadmin@netbox-test:/opt/netbox/netbox$   



nisadmin@netbox-test:/opt/netbox/netbox$ python3 manage.py migrate
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/dist-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/dist-packages/django/core/management/__init__.py", line 420, in execute
    django.setup()
  File "/usr/local/lib/python3.8/dist-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.8/dist-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/lib/python3.8/dist-packages/django/apps/config.py", line 228, in create
    import_module(entry)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'cacheops'

**Reply by candlerb** [reply]
```
$ python3 manage.py sqlmigrate extras 0054
Traceback (most recent call last):
File "manage.py", line 10, in
execute_from_command_line(sys.argv)
File "/usr/local/lib/python3.8/dist-packages/django/core/management/init.py", line 446, in execute_from_command_line
utility.execute()
```

The reason that doesn't work is because you're not running inside a python virtualenv - and it's confusing you because there happens to be a global version of django which has been picked up.  (You don't need *any* django package installed globally for Netbox to work).

The correct way would be:

```
cd /opt/netbox
. venv/bin/activate
python netbox/manage.py ...etc...
```

As for the database ownership problem, you'll have to change the ownership of the affected table or sequence manually.  See e.g. https://github.com/netbox-community/netbox/discussions/6231#discussioncomment-644230 and https://groups.google.com/g/netbox-discuss/c/7IN6rAXJPHA/m/8Qq7H76TAQAJ and [this SO post](https://stackoverflow.com/questions/64322843/sequence-must-have-same-owner-as-table-it-is-linked-to).

**Reply by dayonoiki** [reply]
Hi,
   just found I was in the wrong directory  to run python3 manage.py sqlmigrate extras 0054 . does the output show any issues ?  with regards the database ownership problem, i'll refer this to our database team to have a look. 
[sqlmigrate0054.txt](https://github.com/netbox-community/netbox/files/9085012/sqlmigrate0054.txt)


Regards,

**Reply by candlerb** [reply]
Sorry, the output of sqlmigrate isn't very helpful to me.

Can you just go into a postgres shell (`sudo -u postgres psql netbox`) and run `\d` 

For me, all the owners are the same:

```
                              List of relations
 Schema |                      Name                      |   Type   |  Owner
--------+------------------------------------------------+----------+---------
 public | auth_group                                     | table    | netbox
 public | auth_group_id_seq                              | sequence | netbox
 public | auth_group_permissions                         | table    | netbox
 public | auth_group_permissions_id_seq                  | sequence | netbox
 public | auth_permission                                | table    | netbox
 public | auth_permission_id_seq                         | sequence | netbox
...etc
```

If it's not like that, then I think you need to fix it.  You could do this either by editing the 1.9.x psql dump file (which is just plain text) prior to restoring it, or by suitable "ALTER TABLE/SEQUENCE ..." commands on the restored database.

**Reply by dayonoiki** [reply]
already sent the output  [dt_ds-log.txt](https://github.com/netbox-community/netbox/files/9084107/dt_ds-log.txt) 

but here is the output of \d 
[sql_output.txt](https://github.com/netbox-community/netbox/files/9085413/sql_output.txt)

**Reply by candlerb** [reply]
> already sent the output [dt_ds-log.txt](https://github.com/netbox-community/netbox/files/9084107/dt_ds-log.txt)

Sorry, missed that first time.

It appears you have exactly the problem that was in the postings I linked to before: some tables and/or sequences owned by "postgres" and some by "netbox".  This is probably due to how the system was first installed.

You need to change everything which is currently owned by "postgres" to "netbox" (also check that in configuration.py, you are connecting to the database as the "netbox" user)

**Reply by dayonoiki** [reply]
Hi,
  I was able to change all the tables owned by "postgres" to "netbox" successfully. then ran the the upgrade.sh and restarted services. Netbox now successfully running 2.11.12. And then finally upgraded to 3.2.5 successfully. 

Thanks for your help
